internal/data/data_manager.go

```diff
@@ -52,8 +56,28 @@ type dataManager struct {
 	exchange          dataexchange.Plugin
 	validatorCache    *ccache.Cache
 	validatorCacheTTL time.Duration
+	messageCache      *ccache.Cache
+	messageCacheTTL   time.Duration
 }
 
+type messageCacheEntry struct {
+	msg  *fftypes.Message
+	data []*fftypes.Data
+	size int64
+}
+
+func (mce *messageCacheEntry) Size() int64 {
+	return mce.size
+}
+
+type CacheReadOption int
```

Comment:
Might need some comments on what this is for, unless I missed them elsewhere in the code?

Reply:
I got the explanation from the PR description, but it should probably make its way into the code too.